src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -0,0 +1,99 @@
+import { createElement } from 'preact';
+import { useEffect, useRef } from 'preact/hooks';
+import propTypes from 'prop-types';
+
+function isElementVisible(element) {
+  return element.offsetParent !== null;
+}
+
+/**
+ * Menu wrapper component to facilitate keyboard navigation of a list of
+ * <MenuItem> components.
+ *
+ * Note that `ArrowRight` shall be handled by the <MenuItem> directly and
+ * all other focus() related  navigation is handled here.
+ */
+export default function MenuKeyboardNavigation({
+  className,
+  closeMenu,
+  children,
+  isVisible,
+}) {
+  const menuRef = useRef(null);
+
+  useEffect(() => {
+    if (isVisible) {
+      // TODO: figure out something better than this timeout hack if possible.
+      // The focus won't work without delaying rendering.
+      setTimeout(() => {
+        const firstItem = menuRef.current.querySelector('[role^="menuitem"]');
+        if (firstItem) {
+          firstItem.focus();
+        }
+      });
+    }
+  }, [isVisible]);
+
+  const onKeyDown = event => {
+    const menuItems = Array.from(
+      menuRef.current.querySelectorAll('[role^="menuitem"]')
+    ).filter(isElementVisible);
+
+    let focusedIndex = menuItems.findIndex(el =>
+      el.contains(document.activeElement)
+    );
+
+    let handled = false;
+
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'Esc':
```

Comment:
The `event.key` value is "Escape" rather than "Esc" isn't it?

Reply:
indeed! this was never working